drv/lpc55-update-server/src/main.rs

```diff
@@ -36,13 +36,15 @@ enum UpdateState {
 }
 
 struct ServerImpl {
+    block0: Option<[u8; BLOCK_SIZE_BYTES]>,
```

Comment:
This is only 512 bytes more but how close are we to running out of stack space?

Reply:
Great question! How can I tell?

Reply:
`humility stackmargin`. For reference on my xpressoboard without this change
```
ID TASK                STACKBASE  STACKSIZE   MAXDEPTH     MARGIN
 4 update_server      0x20007e80       2048        968       1080
```
512 bytes would eat into a good bit of our margin. It's certainly not terrible but something to be mindful of.

Reply:
TIL! Thank you.

Reply:
This kind of concern is why I was thinking of writing the header containing block with MAGIG ^ 0xFFFF_FFFF.
In that case, stage0 could also(someday) recognize a disabled image but still compute its hash if needed.

Reply:
I agree that's the preferred approach that we should take in the future.

Reply:
Update on this. Moving the array into self did cause a stack overflow. I'm playing around still.

Reply:
ID TASK                STACKBASE  STACKSIZE   MAXDEPTH     MARGIN
 2 update_server      0x20007ba0       2048       1552        496